### PR TITLE
Update womtool graph and add WDL 1.0 support [BA-6142]

### PR DESCRIPTION
**Pull request by cjllanwarne**
Replacing #5318 which wasn't updating with new changes

### CODE REVIEW

File: wom/src/main/scala/wom/views/GraphPrint.scala [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  implicit val nodeAndLinkMonoid: Monoid[NodesAndLinks] = cats.derived.MkMonoid[NodesAndLinks]
+
+  /*
+   * NB: We don't care that the counter is "atomic", this is just a way to have a "reference-to-a-mutable-Int" to pass around
+   */
+  private def listAllGraphNodes(graph: Graph,
+                                clusterCounter: AtomicInteger,
+                                availableScatterVariables: Map[ScatterVariableNode, DotScatterVariableNode]): NodesAndLinks = {
+
+    graph.nodes.toList.filter(worthDisplaying).foldMap {
```
Comment (mcovarr) [inline wom/src/main/scala/wom/views/GraphPrint.scala]: Brilliant filter name 👏

**Comment by cjllanwarne** [comment]
@mcovarr: 
> how do these changes enable WDL 1.0 support

It replaces the use of WDL draft 2 objects to build a graph with the use of WOM objects to build the graph.

> or the tests confirm that support has been added?

The tests make sure that the examples in the `womtool validate` test suite (which includes WDL draft-2 and 1.0) also run to completion in `womtool graph`. It doesn't assert that the output is _correct_ per se, but it does check that the process exits with a non-failure.

**Comment by vdauwera** [comment]
Thanks everyone! FWIW I would need to have this released by mid-Jan probably so that we can have all the right versions etc show up in the relevant book chapters. Will that be possible?

File: wom/src/main/scala/wom/views/GraphPrint.scala [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+import wom.graph.expression.ExpressionNode
+import wom.graph._
+import wom.types.WomType
+import wom.views.GraphPrint._
+
+final class GraphPrint(executableCallable: ExecutableCallable) {
+
+  def dotString = WorkflowDigraph(executableCallable.name, listAllGraphNodes(executableCallable.graph, new AtomicInteger(0), Map.empty)).dotString
+
+  implicit val nodeAndLinkMonoid: Monoid[NodesAndLinks] = cats.derived.MkMonoid[NodesAndLinks]
```
Comment (ruchim) [inline wom/src/main/scala/wom/views/GraphPrint.scala]: possibly a note for what this is doing? I'm okay looking at magic, but a hint would go a long way :)

File: wom/src/main/scala/wom/views/GraphPrint.scala [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    NodesAndLinks(
+      nodes = Set(DotConditionalNode(id, innerGraphNodesAndLinks.nodes ++ conditionalExpressionNodesAndLinks.nodes)) ,
+      links = innerGraphNodesAndLinks.links ++ conditionalExpressionNodesAndLinks.links
+    )
+  }
+
+}
+
+object GraphPrint {
```
Comment (ruchim) [inline wom/src/main/scala/wom/views/GraphPrint.scala]: some serious dot stuff going on here!

File: wom/src/main/scala/wom/views/GraphPrint.scala (hunk 2 above)
Comment (cjllanwarne) [reply wom/src/main/scala/wom/views/GraphPrint.scala]: I added a comment. This line is just telling the `cats` library that `NodesAndLinks` instances can be added together. It doesn't do anything active, but allows the use of `foldMap` later on in the file.

**Pull request merged**